src/components/App.js

```diff
@@ -4,17 +4,27 @@ import classNames from 'classnames';
 import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
 import Fullscreen from 'react-full-screen';
 import { I18nextProvider } from 'react-i18next';
+import { isNull, isUndefined, isEmpty } from 'lodash';
 import WorkspaceControlPanel from '../containers/WorkspaceControlPanel';
 import Workspace from '../containers/Workspace';
 import WorkspaceAdd from '../containers/WorkspaceAdd';
 import ns from '../config/css-ns';
 import i18n from '../i18n';
+import ErrorDialog from '../containers/ErrorDialog';
 
 /**
  * This is the top level Mirador component.
  * @prop {Object} manifests
  */
 export class App extends Component {
+  /**
+   *
+   * @param {Object} error the object to check
+   */
+  static hasError(error) {
+    return !isNull(error) && !isUndefined(error) && !isEmpty(error);
```

Comment:
I think the error rendering logic shout not be handled  by App. Rather App should simply render the ErrorDialog without any prop. Everything else should be done by the ErrorDialog. The way it is now the error logic is spread over two places.